pandas/core/computation/scope.py

```diff
@@ -127,19 +135,23 @@ def __init__(
             # shallow copy here because we don't want to replace what's in
             # scope when we align terms (alignment accesses the underlying
             # numpy array of pandas objects)
-            self.scope = self.scope.new_child((global_dict or frame.f_globals).copy())
+            self.scope = DeepChainMap(
+                self.scope.new_child((global_dict or frame.f_globals).copy())
+            )
             if not isinstance(local_dict, Scope):
-                self.scope = self.scope.new_child((local_dict or frame.f_locals).copy())
+                self.scope = DeepChainMap(
+                    self.scope.new_child((local_dict or frame.f_locals).copy())
+                )
         finally:
             del frame
 
         # assumes that resolvers are going from outermost scope to inner
         if isinstance(local_dict, Scope):
             resolvers += tuple(local_dict.resolvers.maps)
-        self.resolvers = DeepChainMap(*resolvers)
-        self.temps = {}
+        self.resolvers = DeepChainMap(*resolvers)  # type: DeepChainMap
+        self.temps = {}  # type: Mapping
```

Comment:
Mapping -> Dict

and then resolve `error: Unsupported operand types for + ("List[Dict[Any, Any]]" and "List[Mapping[Any, Any]]")` in full_scope either using return DeepChainMap(self.temps, *self.resolvers.maps, *self.scope.maps) or list.append to create the maps list and then remove assert in add_tmp.

Reply:
will give this a try